client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/RssShuffleManagerBase.java

```diff
@@ -727,7 +727,9 @@ public MutableShuffleHandleInfo reassignOnBlockSendFailure(
       internalHandle =
           (MutableShuffleHandleInfo) ((StageAttemptShuffleHandleInfo) handleInfo).getCurrent();
     }
-    assert internalHandle != null;
+    if (internalHandle == null) {
+      throw new RssException("Unexpected error of the null of internal handle that should not happen");
```

Comment:
An unexpected error occurred: internalHandle should not be null
or 
An unexpected error occurred: internalHandle is null, which should not happen

Reply:
follow your thought.